pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -54,11 +55,15 @@ def on_train_batch_start(
         batch_idx: int,
         unused: Optional[int] = 0,
     ) -> None:
+        if not trainer.logger:
+            raise MisconfigurationException("Cannot use DeviceStatsMonitor callback with Trainer that has no logger.")
```

Comment:
If you add this, you can remove the `assert trainer.logger is not None`s in this file

Reply:
```suggestion
        if not trainer.logger:
            raise MisconfigurationException("Cannot use `DeviceStatsMonitor` callback with `Trainer(logger=False)`.")
```